pkg/util/fs.go

```diff
@@ -77,39 +76,75 @@ func (h *fs) Exists(file string) bool {
 	return err == nil
 }
 
-// Copy copies a set of files from sourcePath to targetPath
-func (h *fs) Copy(sourcePath string, targetPath string) error {
-	if _, err := os.Stat(sourcePath); err != nil {
+// Copy copies the source file to a destination directory.
+// If the destination is a filename, the file will be replaced.
+// If the source is directory, the directory itself and also the content of the
+// source directory will be copied into a target directory.
+func (h *fs) Copy(source string, dest string) (err error) {
+	sourcefile, err := os.Open(source)
+	if err != nil {
 		return err
 	}
+	sourceinfo, err := os.Stat(source)
+	if err != nil {
+		return err
+	}
+	defer sourcefile.Close()
 
-	info, err := os.Stat(targetPath)
-
-	if err != nil || (info != nil && !info.IsDir()) {
-		err = os.Mkdir(targetPath, 0700)
+	if sourceinfo.IsDir() {
+		glog.V(5).Infof("D %q -> %q", source, dest)
+		err = os.MkdirAll(dest, sourceinfo.Mode())
 		if err != nil {
 			return err
 		}
+		return h.CopyContents(source, dest)
+	}
+
+	destfile, err := os.Create(dest)
```

Comment:
yeah I just hit that problem with integration test... it will not work :-/ you have to specify the target file as well. The CopyContents is doing that, the Copy() does not. However, we use that only in onbuild right now and I already fixed that (I specify the dest as full path to a file)